src/Components/Onboarding/Hooks/useOnboardingTracking.ts

```diff
@@ -0,0 +1,78 @@
+import { useTracking } from "react-tracking"
+import {
+  ActionType,
+  ContextModule,
+  StartedOnboarding,
+  CompletedOnboarding,
+  OnboardingUserInputData,
+} from "@artsy/cohesion"
+import { useCallback } from "react"
+
+//TODO: change naming of funcs
+
+export const useOnboardingTracking = () => {
+  const { trackEvent } = useTracking()
+
+  const userStartedOnboarding = useCallback(() => {
+    const payload: StartedOnboarding = {
+      action: ActionType.startedOnboarding,
+    }
+
+    trackEvent(payload)
+  }, [trackEvent])
+
+  const trackQuestionOne = useCallback(
```

Comment:
Worth noting that `useCallback` on web is a performance optimization. If we're not seeing performance bottlenecks, then `useCallback` falls generally under the umbrella of premature optimization. 

(This is slightly different on mobile where every bit of performance matters -- noticed you adding this @laurabeth. But on web, its a need-to basis, otherwise it just adds unnecessary complexity.)

Reply:
But its indeed harmless to add here either way.

Reply:
I think using the useCallback() here solves the problem that I'm seeing of multiple analytics calls firing at once, anywhere between 8-10 of the same call firing and its a bit strange?

Reply:
not a performance bottleneck, but strange behavior overall

Reply:
I read that the rule of thumb here is that internal functions don't need useCallback, but handlers that are passed through to child components (excepting state handlers), or those that will be leveraged in dependency arrays, should be wrapped to avoid unnecessary rerenders. This situation seems to warrant it. 

I will try to find a citation for this so I'm not just blowing smoke. 😤

Reply:
That reasoning seems correct @laurabeth - so the issue is the over-rendering. Good call with the `useCallback` then (and this is indeed a performance thing -- we don't want multiple renders trigging multiple function calls!)

Reply:
I think the actual fix here (correct me if I'm wrong @TMcMeans) was also linked to the placement/trigger of the tracking call within the component. But, with hooks like this it always seems best to just optimize from the outset. That way if someone comes in and either uses the hook w/o check contents we minimize risk, or if they do look at the code and add to it via copy/pasta we're setting the best possible precedent. 🪙 🪙 <- 2 cents